Dockerfile.okd

```diff
@@ -1,5 +1,5 @@
 
-FROM registry.ci.openshift.org/openshift/release:golang-1.16 AS builder
+FROM registry.ci.openshift.org/openshift/release:golang-1.17 AS builder
```

Comment:
Note: it looks like this is needed as the okd-related checks were failing with the following message:

```
[1/2] STEP 5/5: RUN make osbs-build
./build/build.sh
building marketplace-operator...
# sigs.k8s.io/json/internal/golang/encoding/json
vendor/sigs.k8s.io/json/internal/golang/encoding/json/encode.go:1249:12: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
vendor/sigs.k8s.io/json/internal/golang/encoding/json/encode.go:1255:18: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
```